### PR TITLE
Unified account - Review and update connect dApp

### PR DESCRIPTION
**Pull request by Thiendekaco**

### CODE REVIEW

**Review by S2kael — APPROVED** [review]

File: packages/extension-koni-ui/src/Popup/Settings/Security/ManageWebsiteAccess/Detail.tsx [hunk 1]
```diff
@@ -29,27 +30,37 @@ type WrapperProps = ThemeProps;
...
-    const accountListWithoutAll = accounts.filter((opt) => opt.address !== 'ALL');
-
-    return accountAuthTypes.reduce<AccountJson[]>((list, accountAuthType) => {
-      if (accountAuthType === 'evm') {
-        accountListWithoutAll.forEach((account) => account.chainType === 'ethereum' && list.push(account));
-      } else if (accountAuthType === 'substrate') {
-        accountListWithoutAll.forEach((account) => account.chainType === 'substrate' && list.push(account));
-      } else if (accountAuthType === 'ton') {
-        accountListWithoutAll.forEach((account) => account.chainType === 'ton' && list.push(account));
+  const accountProxyItems = useMemo(() => {
```
Comment (lw-cdm) [inline packages/extension-koni-ui/src/Popup/Settings/Security/ManageWebsiteAccess/Detail.tsx:53]: use checkAccountAddressValid

File: packages/extension-koni-ui/src/Popup/Settings/Security/ManageWebsiteAccess/Detail.tsx [hunk 2]
```diff
@@ -117,45 +128,50 @@ function Component ({ accountAuthTypes, authInfo, className = '', goBack, origin
...
-            const newMap = { ...prevMap };
-
-            delete newMap[item.address];
-
-            return newMap;
-          });
-        });
+      item.accounts.forEach((account) => {
+        if (checkAccountAddressValid(account.chainType, authInfo.accountAuthTypes)) {
+          changeAuthorizationPerAccount(account.address, !isEnabled, origin, updateAuthUrls)
```
Comment (lw-cdm) [inline packages/extension-koni-ui/src/Popup/Settings/Security/ManageWebsiteAccess/Detail.tsx:143]: use multi addresses, then send to background at one

File: packages/extension-koni-ui/src/Popup/WalletConnect/ConnectionDetail.tsx [hunk 3]
```diff
@@ -210,13 +210,13 @@ const Component: React.FC<ComponentProps> = (props) => {
           </MetaInfo.Default>
         </MetaInfo>
         <div className='total-account'>
-          {t('{{number}} account connected', { replace: { number: accountItems.length } })}
+          {t('{{number}} account connected', { replace: { number: accountProxyItems.length } })}
         </div>
         <SwList.Section
           className='account-list'
           displayRow
```
Comment (lw-cdm) [inline packages/extension-koni-ui/src/Popup/WalletConnect/ConnectionDetail.tsx:217]: Remove displayRow

File: packages/extension-koni-ui/src/components/AccountProxy/AccountProxyItem.tsx [hunk 4]
```diff
@@ -72,7 +72,7 @@ const AccountProxyItem = styled(Component)<Props>(({ theme }) => {
     paddingRight: token.paddingSM,
     borderRadius: token.borderRadiusLG,
     alignItems: 'center',
-    display: 'flex',
+    display: 'flex !important',
```
Comment (lw-cdm) [inline packages/extension-koni-ui/src/components/AccountProxy/AccountProxyItem.tsx:75]: remove !important

File: packages/extension-koni-ui/src/utils/walletConnect/index.ts [hunk 5]
```diff
@@ -50,16 +50,16 @@ export const getWCAccountList = (accounts: AccountJson[], namespaces: SessionTyp
     rawMap[address] = address;
```
Comment (lw-cdm) [inline packages/extension-koni-ui/src/utils/walletConnect/index.ts:50]: convert to list

File: packages/extension-koni-ui/src/utils/walletConnect/index.ts [hunk 6]
```diff
@@ -50,16 +50,16 @@ export const getWCAccountList = (accounts: AccountJson[], namespaces: SessionTyp
...
   });
 
-  const convertMap: Record<string, AccountJson> = {};
-  const convertList = Object.keys(rawMap).map((address): AccountJson | null => {
-    return findAccountByAddress(accounts, address);
+  const convertMap: Record<string, AccountProxy> = {};
```
Comment (lw-cdm) [inline packages/extension-koni-ui/src/utils/walletConnect/index.ts:53]: refactor to accountProxies.filter

File: packages/extension-koni-ui/src/components/WalletConnect/ConnectionItem.tsx [hunk 7]
```diff
@@ -33,9 +33,9 @@ const Component: React.FC<Props> = (props: Props) => {
 
   const logoUrl = `https://icons.duckduckgo.com/ip2/${domain}.ico`;
 
-  const { accounts } = useSelector((state) => state.accountState);
+  const { accountProxies } = useSelector((state) => state.accountState);
```
Comment (lw-cdm) [inline packages/extension-koni-ui/src/components/WalletConnect/ConnectionItem.tsx:36]: const accountProxies = useSelector((state) => state.accountState.accountProxies);

File: packages/extension-koni-ui/src/components/Layout/parts/ConnectWebsiteModal.tsx [hunk 8]
```diff
@@ -37,21 +38,41 @@ function Component ({ authInfo, className = '', id, isBlocked = true, isNotConne
...
+        const listAddress = accountProxy.accounts.map(({ address }) => address);
+
+        listAddress.forEach((address) => {
+          let addressIsValid = false;
+
+          if (isEthereumAddress(address) && authInfo?.accountAuthTypes.includes('evm')) {
+            addressIsValid = true;
+          }
+
+          if (isSubstrateAddress(address) && authInfo?.accountAuthTypes.includes('substrate')) {
```
Comment (lw-cdm) [inline packages/extension-koni-ui/src/components/Layout/parts/ConnectWebsiteModal.tsx:61]: else if

File: packages/extension-koni-ui/src/components/Layout/parts/ConnectWebsiteModal.tsx [hunk 9]
```diff
@@ -37,21 +38,41 @@ function Component ({ authInfo, className = '', id, isBlocked = true, isNotConne
...
+            addressIsValid = true;
+          }
+
+          if (isTonAddress(address) && authInfo?.accountAuthTypes.includes('ton')) {
```
Comment (lw-cdm) [inline packages/extension-koni-ui/src/components/Layout/parts/ConnectWebsiteModal.tsx:65]: else if

File: packages/extension-koni-ui/src/Popup/WalletConnect/ConnectionDetail.tsx [hunk 10]
```diff
@@ -51,15 +51,15 @@ const Component: React.FC<ComponentProps> = (props) => {
   const { activeModal, inactiveModal } = useContext(ModalContext);
 
   const { chainInfoMap } = useSelector((state) => state.chainStore);
-  const { accounts } = useSelector((state) => state.accountState);
+  const { accountProxies } = useSelector((state) => state.accountState);
```
Comment (lw-cdm) [inline packages/extension-koni-ui/src/Popup/WalletConnect/ConnectionDetail.tsx:54]: accountProxies

File: packages/extension-koni-ui/src/utils/walletConnect/index.ts [hunk 11]
```diff
@@ -40,26 +38,19 @@ export const chainsToWalletConnectChainInfos = (chainMap: Record<string, _ChainI
   });
 };
 
-export const getWCAccountList = (accounts: AccountJson[], namespaces: SessionTypes.Namespaces): AccountJson[] => {
-  const rawMap: Record<string, string> = {};
+export const getWCAccountProxyList = (accountProxies: AccountProxy[], namespaces: SessionTypes.Namespaces): AccountProxy[] => {
+  const rawMap: string[] = [];
```
Comment (lw-cdm) [inline packages/extension-koni-ui/src/utils/walletConnect/index.ts:42]: rename

**Pull request merged**